gix-commitgraph/src/file/init.rs

```diff
@@ -260,7 +260,7 @@ impl TryFrom<&Path> for File {
 // Copied from gix-odb/pack/index/init.rs
 fn read_fan(d: &[u8]) -> ([u32; FAN_LEN], usize) {
     let mut fan = [0; FAN_LEN];
-    for (c, f) in d.chunks(4).zip(fan.iter_mut()) {
+    for (c, f) in d.chunks_exact(4).zip(fan.iter_mut()) {
```

Comment:
Let's keep using `chunks` here, as that will cause a panic which is better than silently working with a `fan` that isn't fully initialised. Please apply this to all other instances of this function as well.

Reply:
`.chunks()` would not catch cases of broken input when when it's multiple of 4, e.g. 8-byte long `d` initializes 2 bytes, and keeps 254 bytes zeroed. I've added an assert instead, which catches all length mismatches.